packages/admin/cms-admin/tsconfig.json

```diff
@@ -3,7 +3,11 @@
     "compilerOptions": {
         "outDir": "lib",
         "rootDir": "src",
-        "strict": true
+        "strict": true,
+        "types": [
+            "@comet/admin-theme",
+            "jest"
+        ]
```

Comment:
This is necessary for the MUI type overrides in admin-theme to take effect in cms-admin. Otherwise you get a type error when trying to use the highlight color

Reply:
I tried around a bit and in our projects (and starter) we should be able to use the more flexible config
```json
"typeRoots": ["node_modules/@types", "node_modules/@comet/admin-theme"],
```
This automatically includes all `@types` packages so you don't have to explicitly add things like `node` or `jest` to the tsconfig.

---

In the mono repo, `typeRoots` don't work. It causes following error:
```
error TS2688: Cannot find type definition file for 'node_modules'.
  The file is in the program because:
    Entry point for implicit type library 'node_modules'

error TS2688: Cannot find type definition file for 'src'.
  The file is in the program because:
    Entry point for implicit type library 'src'
```
According to my research, the problem is our workspace setup: The whole admin-theme folder is symlinked into node_modules. `typeRoots` expects each directory in a given package to contain a `d.ts` file. However, `node_modules` (inside `admin-theme`) and `src` don't contain a `d.ts` file, causing this error (see https://github.com/ui5-community/generator-ui5-ts-app-fcl/issues/5#issuecomment-1199967459 for more info).

Defining `"typeRoots": ["node_modules/@types", "node_modules/@comet/admin-theme/lib"]` also doesn't work in the mono repo, though I'm not sure why. It causes the error:
```
error TS2688: Cannot find type definition file for 'componentsTheme'.
  The file is in the program because:
    Entry point for implicit type library 'componentsTheme'

error TS2688: Cannot find type definition file for 'utils'.
  The file is in the program because:
    Entry point for implicit type library 'utils'
```
though they actually contain `.d.ts.` files. Explicitly defining `index.ts` for each subfolder also didn't help so maybe it's about the missing `package.json`. But I'm not sure.

Reply:
I don't like that.

have a look what we do here to solve a similar problem:

- https://github.com/vivid-planet/comet-starter/pull/20/files#diff-583d2735993363a30c430470c878afc2c38dd9be7e660a0421aa901d6609ca80
- https://github.com/vivid-planet/comet-starter/blob/main/site/next-env.d.ts

note: afaik ts does import all *.d.ts files in src, so just adding this file in src is enough (we do that in admin)

Reply:
vendors.d.ts is a way better solution. I tested it and it works. I even added a codemod for the v7 upgrade:

https://github.com/vivid-planet/comet-upgrade/pull/10